packages/slice-machine/src/features/onboarding/OnboardingProvider.tsx

```diff
@@ -65,15 +73,19 @@ export const OnboardingProvider = ({
     }
   };
 
-  const getStepIndex = (step: OnboardingStep) => {
-    return steps.findIndex(({ id }) => id === step.id);
+  const getStepIndex = (stepId: OnboardingStepId) => {
+    return steps.findIndex(({ id }) => id === stepId);
   };
 
   const isStepComplete = (step: OnboardingStep) => {
     return stepStatus[step.id] ?? false;
   };
 
-  const completedStepCount = Object.values(stepStatus).filter(Boolean).length;
+  const stepsIds = steps.map(({ id }) => id);
+  const completedStepCount = Object.entries(stepStatus).filter(
+    // filtering out steps that are not part of the current experiment variant
+    ([key, value]) => stepsIds.includes(key as OnboardingStepId) && value,
```

Comment:
❓ do we need the casting here?

Reply:
honestly I didn't know how to do it other way, Object.entries is loosing type information inferred from zod schema :/